packages/react/src/components/editor/JSONEditor.tsx

```diff
@@ -77,25 +80,30 @@ export const JSONEditor: React.FunctionComponent<
     onUnmount,
     collapsibleId,
 }) => {
-    const [isInError, setIsInError] = React.useState(!JSONEditorUtils.validateValue(value || defaultValue));
+    const editorValue = defaultValue || value || '{}';
+    const [isInError, setIsInError] = React.useState(!JSONEditorUtils.validateValue(editorValue));
 
     React.useEffect(() => {
         onMount?.();
 
         return onUnmount;
     }, []);
 
+    React.useEffect(() => {
+        const hasError = !JSONEditorUtils.validateValue(editorValue);
+        setIsInError(hasError);
+    }, [editorValue]);
+
     const handleChange = (json: string) => {
         const hasError = !JSONEditorUtils.validateValue(json);
-
         setIsInError(hasError);
         onChange?.(json, hasError);
     };

```

Comment:
You do almost the same thing in the useEffect and the handleChange method, there is no way to factor this part?